### PR TITLE
chore: export juggler as a standalone folder for browser build

### PR DESCRIPTION
**Pull request by aslushnikov**
This leaves our firefox diff to gecko instrumentation changes only.

Drive-by: rename webkit "src" folder into "embedder".

### CODE REVIEW

File: browser_patches/prepare_checkout.sh [hunk 1]
```diff
@@ -31,18 +31,20 @@ FRIENDLY_CHECKOUT_PATH="";
 CHECKOUT_PATH=""
 PATCHES_PATH=""
 BUILD_NUMBER=""
-PLAYWRIGHT_PATH=""
+WEBKIT_EXTRA_FOLDER_PATH=""
+FIREFOX_EXTRA_FOLDER_PATH=""
 if [[ ("$1" == "firefox") || ("$1" == "firefox/") || ("$1" == "ff") ]]; then
```
Comment (dgozman) [inline browser_patches/prepare_checkout.sh:36]: nit: perhaps it makes sense to share some code between prepare_checkout and export?

File: browser_patches/firefox/juggler/Helper.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
+const uuidGen = Cc["@mozilla.org/uuid-generator;1"].getService(Ci.nsIUUIDGenerator);
```
[the remark this answers is not in the document]
Comment (aslushnikov) [reply browser_patches/firefox/juggler/Helper.js:5]: Done.

File: browser_patches/firefox/juggler/jar.mn [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
```
[the remark this answers is not in the document]
Comment (aslushnikov) [reply browser_patches/firefox/juggler/jar.mn:1]: that's the copyright!

File: browser_patches/prepare_checkout.sh (hunk 1 above)
Comment (aslushnikov) [reply browser_patches/prepare_checkout.sh:36]: i'd keep it for now.

File: browser_patches/export.sh [hunk 4]
```diff
@@ -122,11 +129,15 @@ BASE_REVISION=\"$NEW_BASE_REVISION\"" > $EXPORT_PATH/UPSTREAM_CONFIG.sh
...
-if [[ ("$1" == "webkit") || ("$1" == "webkit/") || ("$1" == "wk") ]]; then
-echo "-- patching WebKit embedders"
-rm -rf $EXPORT_PATH/src/*
-mkdir $EXPORT_PATH/src/Tools
-cp -r Tools/Playwright $EXPORT_PATH/src/Tools/
+if [[ ! -z "${WEBKIT_EXTRA_FOLDER_PATH}" ]]; then
+  echo "-- exporting WebKit embedders"
+  rm -rf $EXPORT_PATH/embedder
+  mkdir -p $EXPORT_PATH/embedder
+  cp -r Tools/Playwright "${WEBKIT_EXTRA_FOLDER_PATH}"
```
[the remark this answers is not in the document]
Comment (aslushnikov) [reply browser_patches/export.sh]: Done

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**